kv-storage/src/rocksdb_kv/mod.rs

```diff
@@ -0,0 +1,300 @@
+
+            checkpoint_db
+                .create_checkpoint(current_db_dir.to_path_buf().as_path().join("db"))
+                .unwrap();
+            checkpoint_db
+                .create_checkpoint(checkpoint_db_dir.to_path_buf().as_path().join("db"))
+                .unwrap();
+        }
+
+        Ok(RocksDB {
+            db: DB::open_default(current_db_dir.to_path_buf().as_path().join("db")).unwrap(),
+            origin_path,
+            current_db_dir,
+            checkpoint_db_dir,
+        })
+    }
+
+    async fn commit_checkpoint(&mut self) -> Result<(), super::Error> {
+        let new_checkpoint_db_dir = Temp::new_dir().unwrap();
+        let checkpoint_db = checkpoint::Checkpoint::new(&self.db).unwrap();
+
+        DB::destroy(&Options::default(), self.origin_path.to_str().unwrap()).unwrap();
+        checkpoint_db
+            .create_checkpoint(self.origin_path.to_str().unwrap())
+            .unwrap();
+        checkpoint_db
+            .create_checkpoint(new_checkpoint_db_dir.to_path_buf().as_path().join("db"))
+            .unwrap();
+        self.checkpoint_db_dir = new_checkpoint_db_dir;
+
+        Ok(())
+    }
+
+    async fn revert_to_latest_checkpoint(&mut self) -> Result<(), super::Error> {
+        let new_current_db_dir = Temp::new_dir().unwrap();
+        let new_checkpoint_db_dir = Temp::new_dir().unwrap();
+        {
+            let new_db =
+                DB::open_default(self.checkpoint_db_dir.to_path_buf().as_path().join("db"))
+                    .unwrap();
+            let checkpoint_db = checkpoint::Checkpoint::new(&new_db).unwrap();
+
+            checkpoint_db
+                .create_checkpoint(new_current_db_dir.to_path_buf().as_path().join("db"))
+                .unwrap();
+            checkpoint_db
+                .create_checkpoint(new_checkpoint_db_dir.to_path_buf().as_path().join("db"))
+                .unwrap();
+        }
+
+        self.db = DB::open_default(new_current_db_dir.to_path_buf().as_path().join("db")).unwrap();
+        self.current_db_dir = new_current_db_dir;
+        self.checkpoint_db_dir = new_checkpoint_db_dir;
+
+        Ok(())
+    }
+
+    async fn insert_or_update(&mut self, key: Hash256, value: &[u8]) -> Result<(), super::Error> {
+        let result = self.db.put(key.as_ref(), value);
+        match result {
+            Ok(_) => Ok(()),
+            Err(_) => Err(super::Error::Unknown("unknown error".to_string())),
+        }
+    }
+
+    async fn remove(&mut self, key: Hash256) -> Result<(), super::Error> {
+        let result = self.db.delete(key.as_ref());
+        match result {
+            Ok(_) => Ok(()),
+            Err(_) => Err(super::Error::NotFound),
+        }
+    }
+
+    async fn get(&self, key: Hash256) -> Result<Vec<u8>, super::Error> {
+        let result = self.db.get(key.as_ref());
+        match result {
+            Ok(v) => Ok(v.unwrap()),
+            Err(_) => Err(super::Error::NotFound),
```

Comment:
use the err information, not just discarding it

Reply:
fix